src/main/java/com/google/api/gax/grpc/ServiceApiSettings.java

```diff
@@ -31,233 +27,234 @@
-    return setRetryableCodesOnAllMethods(Sets.newHashSet(codes));
-  }
+   /**
+    * The channel used to send requests to the service.
+    *
+    * If no channel was set, a default channel will be instantiated, using
+    * the connection settings provided.
+    *
+    * See class documentation for more details on channels.
+    */
+    public ManagedChannel getOrBuildChannel() throws IOException {
+      return channelProvider.getChannel(this.getOrBuildExecutor());
+    }
 
-  /**
-   * Sets the retry codes on all of the methods of the API.
-   */
-  public ServiceApiSettings setRetryableCodesOnAllMethods(Set<Status.Code> retryableCodes) {
-    for (ApiCallSettings settings : allCallSettings) {
-      settings.setRetryableCodes(retryableCodes);
+    /**
+     * The Executor used for channels, retries, and bundling..
+     * If no executor was set, a default executor will be instantiated.
+     */
+    public ScheduledExecutorService getOrBuildExecutor() {
+      return executorProvider.getExecutor();
     }
-    return this;
-  }
 
-  /**
-   * Sets the retry params for all of the methods of the API.
-   */
-  public ServiceApiSettings setRetryParamsOnAllMethods(RetryParams retryParams) {
-    for (ApiCallSettings settings : allCallSettings) {
-      settings.setRetryParams(retryParams);
+    /**
+     * Returns true if the channel should be automatically closed with the API wrapper class.
+     */
+    public boolean shouldAutoCloseChannel() {
+      return channelProvider.shouldAutoClose();
     }
-    return this;
-  }
 
-  /**
-   * Sets the generator name and version for the GRPC custom header.
-   */
-  public ServiceApiSettings setGeneratorHeader(String name, String version) {
-    this.serviceGeneratorName = name;
-    this.serviceGeneratorVersion = version;
-    return this;
-  }
+    /**
+     * Sets the generator name and version for the GRPC custom header.
+     */
+    public Builder setGeneratorHeader(String name, String version) {
+      this.serviceGeneratorName = name;
+      this.serviceGeneratorVersion = version;
+      return this;
+    }
 
-  /**
-   * Sets the client library name and version for the GRPC custom header.
-   */
-  public ServiceApiSettings setClientLibHeader(String name, String version) {
-    this.clientLibName = name;
-    this.clientLibVersion = version;
-    return this;
+    /**
+     * Sets the client library name and version for the GRPC custom header.
+     */
+    public Builder setClientLibHeader(String name, String version) {
+      this.clientLibName = name;
+      this.clientLibVersion = version;
+      return this;
+    }
+
+    /**
+     *  Performs a merge, using only non-null fields
+     */
+    @SuppressWarnings("unused")
+    public Builder applyToAllApiMethods(ApiCallSettings.Builder apiCallSettings) throws Exception {
+      // Subclasses need to override
+      throw new Exception("applyToAllApiMethods is not implemented");
```

Comment:
We could actually support this, by having the child class pass in the list of methods, which we were doing before; we just won't expose a getter for that list any more.